openmdao/docs/theory_manual/total_derivs/total_derivs_theory.rst

```diff
@@ -0,0 +1,227 @@
+.. _theory_total_derivatives:
+
+**********************************
+How Total Derivatives are Computed
+**********************************
+
+This is a comprehensive document that explains the various techniques that can be used to compute total derivatives, along with their strengths and weaknesses.
+The goal of this document is to help you understand how the underlying methods work, and when they are appropriate to apply to your model.
+It is designed to be read in the order it's presented, with later sections assuming understanding of earlier ones.
+
+
+Partial vs. Total Derivatives
+-----------------------------
+Before diving into how OpenMDAO solves for total derivatives, it is important that we clearly define a pair of key terms.
+
+    * Partial Derivative: Derivatives of the outputs or residuals of a single component with respect to that component's inputs.
+    * Total Derivative: Derivative of an objective or constraint with respect to design variables.
+
+If you are numerically approximating the total derivatives, using finite-difference or complex-step, then do not need to worry about partial derivatives.
+However analytic derivatives methods achieve better accuracy and lower computational cost by leveraging partial derivatives to compute total derivatives.
+
+Numerical Approximation for Total Derivatives
+----------------------------------------------
+
+Numerical approximation of total derivatives involves treating the overall model as a single function and approximating its total derivatives by examining the change in outputs as inputs are varied.
+These methods involve taking successive steps in each design variable and them computing derivatives based on changes to the output value.
+Generally speaking, each design variable must be examined separately meaning that there the computational cost of these approaches scale linearly with the increasing size of the design space.
+The primary attraction the numerical approach is that it is relatively simple to implement since no knowledge of the internal workings of the model you are trying to differentiate is required.
+
+The most common numerical approach is to use first order finite differences compute the approximations:
```

Comment:
first order -> first-order; 
compute the approximations -> TO compute the approximations